### ISSUE TITLE
Active instrument window highlights assigned FX mixer channel

### ISSUE THREAD

**Issue by mikobuntu** [LMMS/lmms#117]
Hi I hope my title explains what i am meaning. Anyway as we already know if i select an FX Channel from the instrument/sample and double left click whichever FX channel i have chosen, then the mixer strip in the FX_MIxer becomes highlighted.
An improvement i can see in this would be to have the FX_Mixer highlighted whenever an instrument has been chosen and as i mouse scroll through the FX channels in the instrument/sample the FX_MIxer channel becomes highlighted simultaneously.
So basically every new instrument once loaded in a new song already defaults to FX channel "0" / "master" and it will be highlighted automatically until the user starts to scroll through the other possible channels, then these FX_MIxer channels will become highlighted in sync .. thanks Mikobuntu

**Comment by badosu** [comment]
@mikobuntu @tresf Could you help me refine this proposal's description? I want to understand what's required here.

**Comment by tresf** [comment]
Perhaps "Blink the recently selected FX channel as it is assigned" would be a better description? :)

**Comment by tresf** [comment]
From what I'm reading, he wants the FX Mixer to "highlight" or "alert" or "blink" the channels as the mouse wheel scrolls through them.  The context menu that you've created really mitigates the need for this feature, but they could certainly work well together. :)

**Comment by Spekular** [comment]
@tresf really? i thought he meant that when an instrument is selected (in
song editor or bb editor), the fx channel that it is assigned to is
highlighted? Or is that what you are suggesting too and I misunderstood?
Say i add a sample track and 3xOsc to song editor, create three fx
channels, and assign 3xOsc to fx channel 3. I select my sample track, then
i select 3xOsc again. Channel 3 now blinks/highlights in fx mixer.

**Comment by tresf** [comment]
@Spekular I think we're saying the same thing, no?

**Comment by tresf** [comment]
As far as selecting the FX channel, I'm not sure that is desired.  When you have the instrument side-by-side the FX mixer, you may be making adjustments to a part of the route.  Having it lose focus from your work each time you switch instruments can be annoying.

**Comment by badosu** [comment]
@tresf I think what you are describing and what @mikobuntu described are different things.

You want a little indication of what's the instrument mixer, I think he wants to actively change the focused mixer on the mixer window.

@mikobuntu Please tell me if I am mistaken.

**Comment by tresf** [comment]
> You want a little indication of what's the instrument mixer, I think he wants to actively change the focused mixer on the mixer window.

I think you're right... The word "highlight" can be vague, but he's talking about mimicking the double-click behavior when assignment changes.

This is a usability conversation because if the user was actively working on another FX channel, the act of assigning it takes them out of that context.

**Comment by badosu** [comment]
> but he's talking about mimicking the double-click behavior when assignment changes.

Exactly, try double clicking on the FX LcdSpin (the box with the green number indicating what channel this instrument's assigned). What happens is not a *notice* but an actual *highlight*.

**Comment by tresf** [comment]
> What happens is not a notice but an actual highlight.

Semantics, but I don't like the word "highlight" in this context personally, because it's really selecting of that channel, so it is a bit more than a visual indicator... but of course you know that. :)

**Comment by badosu** [comment]
Oh, sorry about that, I did not notice we were talking the same.

I th

**Comment by tresf** [comment]
> I thought you were suggesting just to blink the channel instead of selecting it. Bear in mind that the window is not actually focused so it may not distract that much. Would require a prototype though.

I was, what I was describing until you pointed out we had different ideas (really I had misinterpreted).  So now I'm on the channel select wagon, at least for now.

My point is that if you are quickly navigating between items in the FX mixer, you may THINK you were on (i.e) Channel FX: 1, but the software automagically switched you to (i.e.) Channel FX: 2 when changing the LED SpinnerBox, so the FX plugin listing you are interacting with has changed without knowledge.  It is minor usability issue though and we should have some people try it out probably before saying it is a terrible idea... I just want to be cognizant of it.

**Comment by Umcaruje** [comment]
@mikobuntu could you draw a mockup for this feature, and further explain it? I'm having a hard time understanding this enhancement.

**Comment by tresf** [comment]
Closing due to lack of involvement by OP.  Please feel free to reopen at your own leisure.
